iothub/service/design.md

```diff
@@ -0,0 +1,119 @@
+# RBAC API Design for IoT hub
+
+## Service Client
+
+```csharp
+/// <summary>
+/// Create ServiceClient using AAD token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">AAD <see cref="TokenCredential"> to authenticate with IoT hub.</param>
```

Comment:
TokenCredential being an abstract class, we should specify that the user needs to pass in an implementation of a TokenCredentail

Reply:
From our ADT client: "The <see cref="TokenCredential"/> implementation which will be used to request for the authentication token."

Reply:
It is obvious as it is an abstract class so I did not add that.